lib/bridge/IrcHandler.js

```diff
@@ -330,15 +335,27 @@ IrcHandler.prototype.onJoin = Promise.coroutine(function*(req, server, joiningUs
     if (joiningUser.isVirtual) {
         return BridgeRequest.ERR_VIRTUAL_USER;
     }
+
+    this.quitDebouncer.onJoin(nick, server);
+
     // get virtual matrix user
     let matrixUser = yield this.ircBridge.getMatrixUser(joiningUser);
     req.log.info("Mapped nick %s to %s", nick, JSON.stringify(matrixUser));
     let matrixRooms = yield this.ircBridge.getStore().getMatrixRoomsForChannel(server, chan);
     let promises = matrixRooms.map((room) => {
-        req.log.info("Joining room %s", room.getId());
-        return this.ircBridge.getAppServiceBridge().getIntent(
-            matrixUser.getId()
-        ).join(room.getId());
+        req.log.info("Joining room %s and setting presence to online", room.getId());
+        let intents = [];
```

Comment:
Given you're defining, pushing then `Promise.all()`ing - I would just inline it:
```js
return Promise.all([
  this.ircBridge...,
  this.ircBridge...,
]);
```